gradle-plugins/src/main/java/com/linkedin/pegasus/gradle/PegasusPlugin.java

```diff
@@ -1991,10 +1991,6 @@ protected void configureRestClientGeneration(Project project, SourceSet sourceSe
   {
     // idl directory for api project
     File idlDir = project.file(getIdlPath(project, sourceSet));
-    if (SharedFileUtils.getSuffixedFiles(project, idlDir, IDL_FILE_SUFFIX).isEmpty())
-    {
-      return;
```

Comment:
will there be downsides for packing an empty jar?

why will you need a jar that is empty? Can your logic assume it empty when it doesn't exist?

Reply:
no downsides or compatibility issues. just some negligible space will be used for PDL only projects to publish empty jars.
We will never use the empty jar. only the jars with IDL/ restmodels will be used by clients.

Reply:
updated PR as per offline discussion @junchuanwang @mchen07